autograder/rest_api/views/group_views.py

```diff
@@ -214,7 +217,7 @@ def get(self, *args, **kwargs):
     def patch(self, request, *args, **kwargs):
         group = self.get_object()
 
-        update_data = dict(request.data)
+        update_data = clean_extended_due_dates(dict(request.data), group)
```

Comment:
Once you make the changes to clean_extended_due_date we talked about, theoretically you can remove the call to dict around request.data